code/modules/games/cards.dm

```diff
@@ -55,8 +59,16 @@
 /obj/item/weapon/deck/attackby(obj/O as obj, mob/user as mob)
 	if(istype(O,/obj/item/weapon/hand))
 		var/obj/item/weapon/hand/H = O
-		if(H.parentdeck == src)
+		if(decklimit && (decklimit <= cards.len))
```

Comment:
Can you have a deck with 0 cards? If not, then the first condition here is kinda meaningless.